### DISCUSSION TITLE
Changing snapcast group volume unmutes all clients

### DISCUSSION THREAD

**Discussion by chatziko** [music-assistant/support#2365] [Feature requests and ideas]
I have a snapcast group with 3 speakers (clients), say A,B,C. Often I want to temporarily mute C while A,B are still playing.

Now while C is muted, in snapweb I can still change the volume of the group, which changes the volumes of all clients A,B,C, but it __does not__ unmute C. This makes sense, changing the group volume does not show an intention to unmute any particular client. If, on the other hand, I change the volume of C, it is automatically unmuted, which again makes sense, since I am interacting with C directly.

In MA, however, the behaviour is different: changing the volume of a snapcast group immediately __unmutes all members of the group__! Is this intentional or just a bug? I don't see a benefit in this behaviour, while the risk of unexpectedly blasting a quiet room with music is clear.

**Comment by OzGav** [comment]
Probably not intentional so I’ll change this to a feature request

**Comment by SantiagoSotoC** [comment]
This is implemented this way because at the moment the MA ui does not have a way to manage muting. I understand that it is not ideal, I will see how it can be improved.

**Comment by chatziko** [comment]
In general the link between MA volumes and snapcast volumes seems strange to me.

I have two snapcast clients, named A, B. I link B to A, so now:

- in MA there are 3 volume sliders, one for the group (named "A", cause A is the "master" speaker), and one for each of A, B.
- in snapcast there is one group and two clients. 
  (note that in snapcast the group volume is computed as the average of the clients, it cannot be set independently).

I start by setting all volumes to 100. Now in MA I change the __group__ volume to 50.

- In MA the group volume becomes 50, the volumes of A,B stay 100.
- But in snapcast, the volume of A stays 100 while B becomes 50! (and the group becomes 75, the average).

This is completely unexpected, why B changed to 50 but not A?

I reset everything to 100 again, and now in MA I change A to 50. Now:
- In snapcast A becomes 50, B stays 100 and the group becomes 75 (the expected behaviour)
- In MA A becomes 50, B stays 100 but the group also stays 100 (so now there is an inconsistency between the MA group and the snapcast group).

Changing B to 50 has similar behaviour as changing A to 50.


PS. Thanks a lot for you work, MA looks super promising, I'm currentlt changing all my music setup to it.

**Reply by OzGav** [reply]
Looks like a bug. Will wait for @SantiagoSotoC to just confirm it (or me when I get home tomorrow) and we can create an issue directly from here for you.

**Reply by SantiagoSotoC** [reply]
I'm working on the volume, I think I have something that works quite a bit better, but I don't like it. 
I'm going to report on the “python-snapcast” library, to see if it's the expected behavior.

[https://github.com/music-assistant/server/tree/enhanced-snapcast-volume-handling](https://github.com/music-assistant/server/tree/enhanced-snapcast-volume-handling)

**Reply by OzGav** [reply]
It’s certainly not consistent with how group player volumes are supposed to work. The group volume should be the average of the children. Changing the group volume should have a proportionate change on the children (eg. Change group by 10 all children change by 10 and average then recalculates)

**Reply by SantiagoSotoC** [reply]
Yes, the main problem is that the values are updated in the ui when using snapweb but not when using the ma interface

**Reply by SantiagoSotoC** [reply]
My solution works, but I don't think it's ideal, I already asked in the library repo, I'm waiting for a reply and I'll update here.
Where can we talk about adding a way to control the mute from the ma ui ?

**Reply by OzGav** [reply]
I think in th #devs-playerproviders channel?

**Reply by SantiagoSotoC** [reply]
It is implemented in the server and in the providers, it is the ui that does not support it.

**Comment by SantiagoSotoC** [comment]
https://github.com/music-assistant/server/pull/1324
I think it's fixed now, I removed the mute logic and fixed the volume.
